tools/crldps/main.go

```diff
@@ -0,0 +1,116 @@
+// crldps generates the list of CRL Distribution Point URIs for a given issuing
+// CA and number of shards. It fetches and validates all of those shards. If
+// it doesn't encounter any errors, it pretty-prints the list of all CRLDPs for
+// disclosure in CCADB.
+package main
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/crl/idp"
+)
+
+func main() {
+	log.SetOutput(os.Stderr)
+
+	fs := flag.NewFlagSet("crldps", flag.ContinueOnError)
+	caPath := fs.String("ca", "", "path to an issuing intermediate CA certificate (required)")
+	numShards := fs.Int("shards", 128, "number of CRL shards issued by the CA")
+	err := fs.Parse(os.Args[1:])
+	if err != nil || len(fs.Args()) != 0 {
+		log.Println("Incorrect command line flags; usage:")
+		fs.PrintDefaults()
+		os.Exit(1)
+	}
+
+	issuer, err := core.LoadCert(*caPath)
+	if err != nil {
+		log.Fatalf("Failed to load issuer certificate from %q: %s", os.Args[1], err)
+	}
+
+	crldpBase := fmt.Sprintf("http://%s.c.lencr.org", strings.ToLower(issuer.Subject.CommonName))
+
+	var (
+		anyErr bool
+		crldps []string
+	)
+	for shard := range *numShards {
+		// We 1-index our CRL shards.
+		crldp := fmt.Sprintf("%s/%d.crl", crldpBase, shard+1)
```

Comment:
You can/should use `path.Join` here:

```suggestion
	for shard := range *numShards {
		// We 1-index our CRL shards.
		crldp := crldpBase
		crldp.Path = path.Join("/", fmt.Sprintf("%d.crl", shard))
```

Reply:
No can do: it strips the double slash after the scheme. But probably can do in combination with using url.URL.